transports/msmq/distributor/configuration.md

```diff
@@ -76,3 +76,87 @@ If self-hosting the endpoint here is the code required to enlist the endpoint wi
 snippet: ConfiguringWorker
 
 Similar to self-hosting, if running NServiceBus prior to Version 6, ensure the `app.config` of the worker contains the `MasterNodeConfig` section to point to the hostname where the distributor process is running.
+
+## Troubleshooting
+
+### Performance issues, throughput is low
+
+#### Maximum concurrency
+
+Make sure that the maximum concurrency level is increased on the distributor. By default, the maximum concurrency level is 1. Also, make sure that the workers have the maximum concurrency level set higher.
+
+#### Prefetching
+
+When your workers are using NServiceBus 6, then there is the ability to configure the worker capacity independently from the maximum concurrency. By assigning a capacity value higher then the maximum concurrency you are able to reduce the latency to start working on a next message as by default the worker first reports it is available for work and then gets a new message forwarded by the distributor.
+
+#### Sender side distribution
+
+Using [sender side distribution](h/transports/msmq/sender-side-distribution) instead of using the distributor might be a better alternative if you are able to use NServiceBus 6 and is much more suitable for scenarios where IO is the bottleneck. 
+
+#### Do not run all distributors on a single box
+
+Distributor endpoints are often deployed on *the distributor machine*. This is one machine that hosts all distributor endpoints. This becomes the bottleneck under load as all messages flow throught this machine. An alternative is to have multiple distributor machines on physically different networks interface cards.
+
+Alternative, when first hosting all endpoints on a single machine do not use the distributor to now run *all* endpoints on multiple workers. First balance your current endpoints by moving endpoints. Eventually you will have only a single endpoint running on a machine. If this machine now becomes your bottleneck then you might first want to upscale this first and if upscaling becomes too costly then make use of the distributor. 
```

Comment:
This section is really not clear to me:

> do not use the distributor to now run *all* endpoints on multiple workers

you mean that when you have the distributor and multiple workers on a single machine then each of the workers should not "run" all endpoints? not sure what's the scenario here...

>First balance your current endpoints by moving endpoints.

Balance... the load? or what? move endpoints where? between workers, i.e. that each worker runs some of the endpoints? can you clarify?

>  Eventually you will have only a single endpoint running on a machine.

How's that related to having many workers? How many workers will you have on each machine and what's preventing you from having one worker run multiple endpoints on a single machine?

> If this machine now becomes your bottleneck then you might first want to upscale this first and if upscaling becomes too costly then make use of the distributor.

I think we should start from this instead of having this at the end, but since this is troubleshooting guide I'm not clear why we consider here the scenario of not having a distributor at all? Maybe would make sense more in the "when to use distributor section" (another page) or maybe after reprhasing it'll be more visible why it's mentioned here.